File: okhttp/src/main/java/com/guoxiaoxing/okhttp/demo/MainActivity.java
package com.guoxiaoxing.okhttp.demo;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;

import java.io.IOException;

import okhttp3.Call;
import okhttp3.Callback;
import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.Response;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {

    private OkHttpClient okHttpClient;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        findViewById(R.id.btn_async_request).setOnClickListener(this);

        okHttpClient = new OkHttpClient.Builder().build();
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.btn_async_request:
                for (int i = 0; i < 100; i++) {
                    asyncRequest("http://www.baidu.com");
                }
                break;
        }
    }

    private void syncRequest(String url) throws IOException {
        Request request = new Request.Builder()
                .url(url)
                .build();
        OkHttpClient okHttpClient = new OkHttpClient.Builder()
                .build();
        Response response = okHttpClient.newCall(request).execute();
    }

    private void asyncRequest(String url) {
        Request request = new Request.Builder()
                .url(url)
                .build();
        okHttpClient.newCall(request).enqueue(new Callback() {
            @Override
            public void onFailure(Call call, IOException e) {
            }

            @Override
            public void onResponse(Call call, Response response) throws IOException {

            }
        });
    }
}

File: okhttp/src/main/java/com/guoxiaoxing/okhttp/demo/MainActivity.kt
package com.guoxiaoxing.okhttp.demo

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import kotlinx.android.synthetic.main.activity_main.*
import okhttp3.*
import java.io.IOException

class MainActivity : AppCompatActivity(), View.OnClickListener {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        btn_send_get_request.setOnClickListener(this)
    }

    override fun onClick(v: View?) {
        when (v?.id) {
            R.id.btn_send_get_request -> getAsynchronization("http://www.baidu.com")
        }
    }


    /**
     * 发送Get请求-异步
     */
    private fun getAsynchronization(url: String) {
        val okhttpClient: OkHttpClient = OkHttpClient.Builder().build()
        val request: Request = Request.Builder()
                .url(url)
                .build()
        okhttpClient.newCall(request).enqueue(object : Callback {

            override fun onFailure(call: Call?, e: IOException?) {
                Log.d(App.TAG, e.toString())
            }

            override fun onResponse(call: Call?, response: Response?) {
                Log.d(App.TAG, response?.body()?.string())
            }
        })
    }

    /**
     * 发送Get请求-同步
     */
    private fun getSynchronization(url: String) {
        val okhttpClient: OkHttpClient = OkHttpClient.Builder().build()
        val request: Request = Request.Builder()
                .url(url)
                .build()
        val response: Response = okhttpClient.newCall(request).execute()
    }
}
